stdlib/public/core/UnicodeHelpers.swift

```diff
@@ -122,6 +122,46 @@ internal func _utf8ScalarLength(_ x: UInt8) -> Int {
   return (~x).leadingZeroBitCount
 }
 
+@inlinable @inline(__always)
+internal func _utf16Length(_ rawBuffer: UnsafeRawBufferPointer) -> Int {
+  guard rawBuffer.count > 0 else { return 0 }
+  _internalInvariant(!UTF8.isContinuation(rawBuffer.first.unsafelyUnwrapped))
+  _internalInvariant(
+       UTF8.isASCII(rawBuffer.last.unsafelyUnwrapped)
+    || UTF8.isContinuation(rawBuffer.last.unsafelyUnwrapped))
```

Comment:
Similarly, `idx` may not be scalar-aligned, in which case the slice may end up starting/ending on partial scalars, violating these invariants.

Reply:
I think in practice it always is for this caller, but I wasn't 100% sure, so I put the invariants in to catch if I was wrong :)